src/anndata/_io/specs/methods.py

```diff
@@ -395,11 +405,23 @@ def read_string_array(d, _reader):
     return read_array(d.asstr(), _reader=_reader)
 
 
+@_REGISTRY.register_read(H5Array, IOSpec("string-array-nullable", "0.1.0"))
+def read_string_array_nullable(d, _reader):
+    return pd.array(read_array(d.asstr(), _reader=_reader), dtype="string")
+
+
 @_REGISTRY.register_read_partial(H5Array, IOSpec("string-array", "0.2.0"))
 def read_string_array_partial(d, items=None, indices=slice(None)):
     return read_array_partial(d.asstr(), items=items, indices=indices)
 
 
+@_REGISTRY.register_read_partial(H5Array, IOSpec("string-array-nullable", "0.1.0"))
```

Comment:
Probably best to stop adding things to `read_partial`

Reply:
I removed them again